new-installer/internal/steps/aws/ec2log.go

```diff
@@ -0,0 +1,210 @@
+// based on variable.tf default definitions.
+func NewDefaultEC2LogVariables() EC2logVariables {
+	return EC2logVariables{
+		S3Prefix:    "",
+		CustomerTag: "",
+	}
+}
+
+type EC2LogStep struct {
+	variables          EC2logVariables
+	backendConfig      TerraformAWSBucketBackendConfig
+	RootPath           string
+	KeepGeneratedFiles bool
+	TerraformUtility   steps.TerraformUtility
+	AWSUtility         AWSUtility
+}
+
+func CreateEC2LogStep(rootPath string, keepGeneratedFiles bool, terraformUtility steps.TerraformUtility, awsUtility AWSUtility) *EC2LogStep {
+	return &EC2LogStep{
+		RootPath:           rootPath,
+		KeepGeneratedFiles: keepGeneratedFiles,
+		TerraformUtility:   terraformUtility,
+		AWSUtility:         awsUtility,
+	}
+}
+
+func (s *EC2LogStep) Name() string {
+	return "EC2LogStep"
+}
+
+func (s *EC2LogStep) Labels() []string {
+	return EC2LOGStepLabels
+}
+
+func (s *EC2LogStep) ConfigStep(ctx context.Context, config config.OrchInstallerConfig, runtimeState config.OrchInstallerRuntimeState) (config.OrchInstallerRuntimeState, *internal.OrchInstallerError) {
+	// no output to check if the EC2Log step is already created. Does it need to be added?
+	s.variables = NewDefaultEC2LogVariables()
+
+	if config.Global.OrchName == "" {
+		return runtimeState, &internal.OrchInstallerError{
+			ErrorCode: internal.OrchInstallerErrorCodeInvalidArgument,
+			ErrorMsg:  "OrchName is not set",
+		}
+	}
+	if config.AWS.Region == "" {
+		return runtimeState, &internal.OrchInstallerError{
+			ErrorCode: internal.OrchInstallerErrorCodeInvalidArgument,
+			ErrorMsg:  "AWS Region is not set",
+		}
+	}
+	if runtimeState.AWS.NodeGroupRole == "" {
+		return runtimeState, &internal.OrchInstallerError{
+			ErrorCode: internal.OrchInstallerErrorCodeInvalidArgument,
+			ErrorMsg:  "NodeGroupRole is not set",
+		}
+	}
+
+	s.variables.Region = config.AWS.Region           // Region is required for AWS resources
+	s.variables.ClusterName = config.Global.OrchName // ClusterName is required for AWS resources
+	s.variables.NodeGroupRole = runtimeState.AWS.NodeGroupRole
+
+	//to do: need to create a default for these 2 variables if not provided
+	s.variables.S3Prefix = runtimeState.DeploymentID
+	s.variables.CustomerTag = config.AWS.CustomerTag
+
+	s.backendConfig = TerraformAWSBucketBackendConfig{
+		Region: config.AWS.Region,
+		Bucket: config.Global.OrchName + "-" + runtimeState.DeploymentID,
+		Key:    EC2LogBackendBucketKey,
+	}
+	return runtimeState, nil
+}
+
+func (s *EC2LogStep) PreStep(ctx context.Context, config config.OrchInstallerConfig, runtimeState config.OrchInstallerRuntimeState) (config.OrchInstallerRuntimeState, *internal.OrchInstallerError) {
+	if config.AWS.PreviousS3StateBucket == "" {
+		// No need to migrate state, since there is no previous state bucket
+		return runtimeState, nil
+	}
+
+	// Need to move Terraform state from old bucket to new bucket:
+	oldEC2LogBucketKey := fmt.Sprintf("%s/ec2log/%s", config.AWS.Region, config.Global.OrchName)
```

Comment:
Use `cluster` since the ec2log was under the cluster module and use the same key

```suggestion
	oldEC2LogBucketKey := fmt.Sprintf("%s/cluster/%s", config.AWS.Region, config.Global.OrchName)
```